ocaml/idl/datamodel.ml

```diff
@@ -5271,7 +5306,8 @@ let network =
           field ~qualifier:DynamicRO ~in_product_since:rel_orlando ~ty:(Map(String, Ref _blob)) ~default_value:(Some (VMap [])) "blobs" "Binary blobs associated with this network";
           field ~writer_roles:_R_VM_OP ~in_product_since:rel_orlando ~default_value:(Some (VSet [])) ~ty:(Set String) "tags" "user-specified tags for categorization purposes";
           field ~qualifier:DynamicRO ~in_product_since:rel_tampa ~default_value:(Some (VEnum "unlocked")) ~ty:network_default_locking_mode "default_locking_mode" "The network will use this value to determine the behaviour of all VIFs where locking_mode = default";
-          field ~qualifier:DynamicRO ~in_product_since:rel_creedence ~default_value:(Some (VMap [])) ~ty:(Map (Ref _vif, String)) "assigned_ips" "The IP addresses assigned to VIFs on networks that have active xapi-managed DHCP"
+          field ~qualifier:DynamicRO ~in_product_since:rel_creedence ~default_value:(Some (VMap [])) ~ty:(Map (Ref _vif, String)) "assigned_ips" "The IP addresses assigned to VIFs on networks that have active xapi-managed DHCP";
+          field ~qualifier:StaticRO ~in_product_since:rel_inverness ~default_value:(Some (VSet [])) ~ty:(Set network_purpose) "purposes" "Purposes for which the network is to be used";
```

Comment:
Maybe this should be `DynamicRO`, because one can modify it with some calls. Another example of such a `RO/untime` field with a custom setter is `subject.roles`. And many of the VDI fields, for example `cbt_enabled`.
However, the [xapi docs](https://xapi-project.github.io/xen-api/basics.html) says this about RO/runtime fields:
> RO/runtime: the field is Read Only. Furthermore, its value is automatically computed at runtime. For example: current CPU load and disk IO throughput.

which is a bit confusing.

I think we cannot set this to RW sadly, because then we'd get auto-generated setters.

Reply:
The main difference between StaticRO and DynamicRO is that the former adds a constructor argument, i.e. to `network.create`. I think that we should use DynamicRO here.

Reply:
Yes, and there's one place already where I'm specifying a purpose (HIMN) in the call to the constructor.

Reply:
Note the distinction between the `network.create` XenAPI call and the function in xapi's (internal) DB interface.

Reply:
(...but if we're only going to have the nbd-related purposes in the Enum for now, then I'll have to undo that.)